lib/mix/lib/mix/tasks/compile.all.ex

```diff
@@ -15,13 +15,9 @@ defmodule Mix.Tasks.Compile.All do
     Mix.Project.build_structure
 
     with_logger_app fn ->
-      res =
-        Enum.map(Mix.Tasks.Compile.compilers(), fn(compiler) ->
-          Mix.Task.run("compile.#{compiler}", args)
-        end)
-
+      res = do_compile(Mix.Tasks.Compile.compilers(), args)
```

Comment:
Can't this be written as:

```elixir
res =
  Enum.map(Mix.Tasks.Compile.compilers(), fn(compiler) ->
    Mix.Task.run("compile.#{compiler}", args)
  end)

res = Mix.Task.Compiler.normalize_results(res)
```

And then we get rid of the `do_compile`?

Reply:
Actually, I see. You want to stop if you get an error. Let me think about this a bit more.

Reply:
This is fine as is, thanks! :)